### PR TITLE
Build new collections for finding trials from "trial_match" directly and match trials for query data

### PR DESCRIPTION
**Pull request by victoria34**
4 newly created collections:
"trial_query": the collection is saved trials that found from collection "trial_match"
"new_trial_query": saved trials for new generated match trial record based on each query
"new_clinical": save newly added clinical data
"new_genomic": save newly added genomic data

### CODE REVIEW

File: matchengine.py [hunk 1]
```diff
@@ -127,18 +127,37 @@ def load_json(self, clinical, genomic):
...
         subprocess.call(cmd2.split(' '))
 
         # convert string to date object
         for clinical_item in self.db.clinical.find():
             for col in ['BIRTH_DATE', 'REPORT_DATE']:
-                if type(clinical_item[col]) is not dt:
-                    clinical_item[col] = dt.datetime.combine(dt.datetime.strptime(str(clinical_item[col]), '%Y-%m-%d'), dt.time.min)
+                if type(clinical_item[col]) is not dt.datetime:
+                    clinical_item[col] = dt.datetime.strptime(str(clinical_item[col]), '%Y-%m-%d')
+                    clinical_item[col] = dt.datetime.strptime(str(clinical_item[col]), '%Y-%m-%d %X')
```
Comment (zhx828) [inline matchengine.py]: ?why additional parsing?

Comment (victoria34) [reply matchengine.py]: I cannot convert date string to date object directly in format `%Y-%m-%d %X`, so I added the additional parsing.

File: matchengine/engine.py [hunk 2]
```diff
@@ -645,6 +686,9 @@ def find_trial_matches(self):
             else:
                 logging.info('Matching trial %s' % trial['nct_id'])
 
+            # skip trials that found in trial_match for query_data
```
Comment (zhx828) [inline matchengine/engine.py]: ？？？

**Review by zhx828 — COMMENTED** [review]
What's the logic to create new_clinical and new_genomic?

File: matchengine/engine.py (hunk 2 above)
Comment (victoria34) [reply matchengine/engine.py]: Sorry, the comment location is wrong. This means that MatchEngine will skip trials that found in matched history before matching.

File: matchengine.py [hunk 3]
```diff
@@ -22,19 +22,23 @@
...
+    def __init__(self, db, uri):
 
         self.db = db
         self.load_dict = {
             'yml': self.yaml_to_mongo,
             'bson': self.bson_to_mongo,
             'json': self.json_to_mongo
         }
+        self.uri = uri
+        self.mlab = False
```
Comment (zhx828) [inline matchengine.py]: I don't think this is the right way to interpret the database location. You should have some figurations and passed in as parameters. Then by default, it goes to localhost unless specified.

File: matchengine/engine.py [hunk 4]
```diff
@@ -262,14 +273,19 @@ def general_match(self, conditions):
                     'CLINICAL_ID': 1,
                     'MMR_STATUS': 1,
                     'ACTIONABILITY': 1,
-                    '_id': 1
+                    '_id': 1,
+                    'ONCOKB_GENOMIC_ID': 1
```
Comment (zhx828) [inline matchengine/engine.py]: Why do we have a oncokb specified genomic id here?

File: matchengine/engine.py [hunk 5]
```diff
@@ -707,17 +755,24 @@ def _assess_match(self, mrn_map, trial_matches, trial, trial_segment, match_segm
                     'VITAL_STATUS': 1,
                     'FIRST_LAST': 1,
                     'GENDER': 1,
-                    '_id': 1
+                    '_id': 1,
+                    'ONCOKB_CLINICAL_ID': 1
```
Comment (zhx828) [inline matchengine/engine.py]: Same question here

File: matchengine/engine.py [hunk 6]
```diff
@@ -42,6 +42,14 @@ def __init__(self, db):
         # get the database.
         self.db = db
 
+        # check if match trial based on newly added clinical and genomic data
+        if self.db.new_genomic.count() > 0 and self.db.new_clinical.count() > 0:
+            self.query = True
+            self.oncokb_matched_results = oncokb_api_match(self.db, "new_genomic")
```
Comment (zhx828) [inline matchengine/engine.py]: Now we changed to annotated_variants, we need a solution to change the match method. OncoKB could be the default, but we need to give a way to change that.

File: tests/test_engine.py [hunk 7]
```diff
@@ -119,6 +119,22 @@ def test_run_query(self):
         assert 'actionability' in matches[0]
         assert matches[0]['mmr_status'] == 'Proficient (MMR-P / MSS)'
 
+        # test a genomic node only has 'annotated_variant' and 'hugo_symbol'
+        # run in oncokb_match()
+        self.add_genomic_annotated_variant()
```
Comment (zhx828) [inline tests/test_engine.py:124]: What are the purposes of these tests?

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: matchengine/engine.py (hunk 4 above)
Comment (victoria34) [reply matchengine/engine.py]: It's used to unique each genomic record. Sometimes genomic records may be duplicate. I'll change the name to `UNIQUE_GENOMIC_ID`.

File: tests/test_engine.py (hunk 7 above)
Comment (victoria34) [reply tests/test_engine.py:124]: This test is used for testing oncokb_match().

**Pull request closed without merging**